bpf/lib/common.h

```diff
@@ -276,27 +275,25 @@ enum {
 
 /* Magic skb->mark markers which identify packets originating from the host
  *
- * The upper 16 bits contain the magic marker values which indicate whether
- * the packet is coming from an ingress or egress proxy, or a local process.
+ * The upper 16 bits contain
+ *  - the magic marker values which indicate whether the packet is coming from
+ *    an ingress or egress proxy, or a local process.
+ *  - the cluster id
  *
  * The lower 16 bits may contain the security identity of the original source
  * endpoint.
  */
-#define MARK_MAGIC_HOST_MASK		0xFFF
-#define MARK_MAGIC_PROXY_INGRESS	0xFEA
-#define MARK_MAGIC_PROXY_EGRESS		0xFEB
-#define MARK_MAGIC_HOST			0xFEC
-#define MARK_IDENTITY_MASK		(0xFFFF << 16)
-
-#define SOURCE_INGRESS_PROXY 1
-#define SOURCE_EGRESS_PROXY 2
+#define MARK_MAGIC_HOST_MASK		0xF00
+#define MARK_MAGIC_PROXY_INGRESS	0xA00
+#define MARK_MAGIC_PROXY_EGRESS		0xB00
+#define MARK_MAGIC_HOST			0xC00
```

Comment:
During upgrade/restore, I believe that this will cause misclassification of traffic. Not sure if there is much we can do other than maybe document that upgrade for this version will not be hitless?

Reply:
Upgrade currently restarts the L7 proxy so the upgrade will quickly interrupt all connections anyway. Regarding host policy, there is a very short possible race condition in between re-creating the iptables rules and running init.sh to update the bpf_netdev.c program. Are you worried about something more?

Reply:
Nope, that's basically it.

Reply:
Only caveat I'm thinking of is that if you have a large number of pods, or a small amount of CPU available, then the 'short possible race condition' may become longer.

Reply:
Given it is bpf_netdev responsible for the translation. The number of endpoints should have minimal impact as the identity is carried outside of the mark afterwards. We could potentially keep the old iptables rule around until the new bpf_netdev is compiled but the race window is so small right now that it feels like we are micro optimizing in this case.